src/main/drivers/bus_spi.c

```diff
@@ -97,6 +97,15 @@ SPIDevice spiDeviceByInstance(SPI_TypeDef *instance)
     return SPIINVALID;
 }
 
+SPI_TypeDef *spiInstanceByDevice(SPIDevice device)
+{
+    if (device < SPIDEV_1 || device > (int)ARRAYLEN(spiHardwareMap)) {
```

Comment:
off-by-one error:
 device >= (int)ARRAYLEN(spiHardwareMap

Also using SPIDEV_1 is slightly confusing, array indexes are <0..ARRAYLEN(spiHardwareMap))